service/prometheus/scrapeconfig.go

```diff
@@ -61,6 +66,31 @@ func GetScrapeConfigs(services []v1.Service, certificateDirectory string) ([]con
 						},
 					},
 				},
+				KubernetesSDConfigs: []*config.KubernetesSDConfig{
+					{
+						APIServer: config.URL{&url.URL{
+							Scheme: httpsScheme,
+							Host:   getTargetName(services[0]),
```

Comment:
This is a bit of a weird hold over from the idea of having multiple services for a guest cluster, as opposed to the newer approach of having one service for the master, and using prometheus' kubernetes service discovery for everything else. Needs to be refactored in future, but I don't know exactly how that looks yet. Will boyscout.

Reply:
And it holds that there will always be one item in the `services` slice at this point, so I'm not gonna invest time checking the length etc., especially as this will be going away shortly.

Reply:
Sounds like this topic deserves a follow up issue we can track separately.

Reply:
Why you can not you use DNS name? `master.XXXXX` should do the job and no logic will be needed if we have 3 masters at some point. But you still need to skip TLS verify, because `master.XXXXX` is not in alternative names.

Reply:
This is the case, we're using `master.XXXXX` here.

Reply:
Tracking refactoring in https://github.com/giantswarm/giantswarm/issues/1482